kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/KafkaProxySampleTest.java

```diff
@@ -0,0 +1,189 @@
+import org.jbpm.services.api.ProcessService;
+import org.jbpm.springboot.samples.events.listeners.CountDownLatchEventListener;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.ToxiproxyContainer;
+
+import static org.apache.kafka.clients.producer.ProducerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.CLIENT_ID_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+
+import static org.jbpm.springboot.samples.KafkaFixture.KAFKA_PROCESS_ID;
+import static org.jbpm.springboot.samples.KafkaFixture.KAFKA_RESULT;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+@SpringBootTest(classes = {JBPMApplication.class, TestAutoConfiguration.class}, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations="classpath:application-test.properties")
+@DirtiesContext(classMode= DirtiesContext.ClassMode.AFTER_CLASS)
+public class KafkaProxySampleTest extends KafkaBaseTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(KafkaProxySampleTest.class);
+    
+    @ClassRule
+    public static final SpringClassRule scr = new SpringClassRule();
+ 
+    @Rule
+    public final SpringMethodRule smr = new SpringMethodRule();
+
+    @Autowired
+    private DeploymentService deploymentService;
+    
+    @Autowired
+    private ProcessService processService;
+    
+    @Autowired
+    CountDownLatchEventListener countDownLatchEventListener;
+    
+    protected static KafkaFixture kafkaFixture = new KafkaFixture();
+    
+    protected String deploymentId;
+    
+    protected String proxyBootstrap;
+
+    protected ToxiproxyContainer.ContainerProxy kafkaProxy;
+    
+    @BeforeClass
+    public static void generalSetup() {
+        kafkaFixture.generalSetup();
+    }
+
+    @Before
+    public void setup() throws IOException, InterruptedException {
+        toxiproxy.start();
+        kafkaProxy = toxiproxy.getProxy(kafka, 9093);
```

Comment:
Not sure if we will have any potential issues when trying to use this port even if it's going through a container. We might get that the port is already in use in some of the jenkins jobs... but probably this port is not mapped to the actual jenkins machine?

Reply:
we should not hardcoded testing port at all, please use a property to set it properly by the environment where it runs from maven.

Reply:
Parameterized as system property inside maven surefire plugin